functions/inputs/from_json.go

```diff
@@ -0,0 +1,193 @@
+	flux.RegisterOpSpec(FromJSONKind, newFromJSONOp)
+	plan.RegisterProcedureSpec(FromJSONKind, newFromJSONProcedure, FromJSONKind)
+	execute.RegisterSource(FromJSONKind, createFromJSONSource)
+}
+
+func createFromJSONOpSpec(args flux.Arguments, a *flux.Administration) (flux.OperationSpec, error) {
+	var spec = new(FromJSONOpSpec)
+
+	if json, ok, err := args.GetString("json"); err != nil {
+		return nil, err
+	} else if ok {
+		spec.JSON = json
+	}
+
+	if file, ok, err := args.GetString("file"); err != nil {
+		return nil, err
+	} else if ok {
+		spec.File = file
+	}
+
+	if spec.JSON == "" && spec.File == "" {
+		return nil, errors.New("must provide json raw text or filename")
+	}
+
+	if spec.JSON != "" && spec.File != "" {
+		return nil, errors.New("must provide exactly one of the parameters json or file")
+	}
+
+	if spec.File != "" {
+		if _, err := os.Stat(spec.File); err != nil {
+			return nil, errors.Wrapf(err, "failed to stat json file: %s", spec.File)
+		}
+	}
+
+	return spec, nil
+}
+
+// FromJSONOpSpec defines the `fromJSON` function signature
+type FromJSONOpSpec struct {
+	JSON string `json:"json"`
+	File string `json:"file"`
+}
+
+func newFromJSONOp() flux.OperationSpec {
+	return new(FromJSONOpSpec)
+}
+
+func (s *FromJSONOpSpec) Kind() flux.OperationKind {
+	return FromJSONKind
+}
+
+// FromJSONProcedureSpec describes the `fromJSON` prodecure
+type FromJSONProcedureSpec struct {
+	plan.DefaultCost
+	JSON string
+	File string
+}
+
+func newFromJSONProcedure(qs flux.OperationSpec, pa plan.Administration) (plan.ProcedureSpec, error) {
+	spec, ok := qs.(*FromJSONOpSpec)
+	if !ok {
+		return nil, fmt.Errorf("invalid spec type %T", qs)
+	}
+	return &FromJSONProcedureSpec{
+		JSON: spec.JSON,
+		File: spec.File,
+	}, nil
+}
+
+func (s *FromJSONProcedureSpec) Kind() plan.ProcedureKind {
+	return FromJSONKind
+}
+
+func (s *FromJSONProcedureSpec) Copy() plan.ProcedureSpec {
+	ns := new(FromJSONProcedureSpec)
+	ns.JSON = s.JSON
+	ns.File = s.File
+	return ns
+}
+
+func createFromJSONSource(prSpec plan.ProcedureSpec, dsid execute.DatasetID, a execute.Administration) (execute.Source, error) {
+	spec, ok := prSpec.(*FromJSONProcedureSpec)
+	if !ok {
+		return nil, fmt.Errorf("invalid spec type %T", prSpec)
+	}
+
+	var jsonReader io.Reader
+
+	if spec.File != "" {
+		f, err := os.Open(spec.File)
+		if err != nil {
+			return nil, err
+		}
+		jsonReader = bufio.NewReaderSize(f, bufferSize)
+	} else {
+		jsonReader = strings.NewReader(spec.JSON)
+	}
+
+	decoder := influxql.NewResultDecoder(a.Allocator())
+	results, err := decoder.Decode(ioutil.NopCloser(jsonReader))
+	if err != nil {
+		return nil, err
+	}
+
+	return &JSONSource{id: dsid, results: results}, nil
+}
+
+type JSONSource struct {
+	results flux.ResultIterator
+	id      execute.DatasetID
+	ts      []execute.Transformation
+}
+
+func (c *JSONSource) AddTransformation(t execute.Transformation) {
+	c.ts = append(c.ts, t)
+}
+
+func (c *JSONSource) Run(ctx context.Context) {
```

Comment:
This is all technically correct, but can we re-tool this to use the SourceDecoder type?  This will be another good example of using to encourage adoption of it:  https://github.com/influxdata/flux/blob/ab9815acfd8e9d429835a39b3e6e8d02cfd7c238/functions/inputs/source_iterator.go#L21

I think @jsternberg had some resistance to implementing that interface and instead went for a pure iterator-type interface for his decoder.  I'd be willing to have a discussion about updating the SourceDecoder to be a more traditional iterator type with  `More()` / `Next()` instead of `fetch`/`decode`

Reply:
@aanthony1243 Could this be done in a follow on PR, since this is more of an internal tool for testing rather than a general one?

Reply:
yeah, we can drop this point for now.